File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-1.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        i = j = 0
        res = len(t)
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                res -= 1
                j += 1
            i += 1
        
        return res

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private ArrayList<Integer> res;
    public List<Integer> preorderTraversal(TreeNode root) {
        res = new ArrayList<>();
        preOrder(root);
        return res;
    }

    private void preOrder(TreeNode node){
        if (node == null){
            return;
        }

        res.add(node.val);
        preOrder(node.left);
        preOrder(node.right);
    }
}